linode/user/framework_models.go

```diff
@@ -0,0 +1,214 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/linode/linodego"
+)
+
+type DataSourceModel struct {
+	Username          types.String `tfsdk:"username"`
+	SSHKeys           types.List   `tfsdk:"ssh_keys"`
+	Email             types.String `tfsdk:"email"`
+	Restricted        types.Bool   `tfsdk:"restricted"`
+	GlobalGrants      types.List   `tfsdk:"global_grants"`
+	DomainGrant       types.Set    `tfsdk:"domain_grant"`
+	FirewallGrant     types.Set    `tfsdk:"firewall_grant"`
+	ImageGrant        types.Set    `tfsdk:"image_grant"`
+	LinodeGrant       types.Set    `tfsdk:"linode_grant"`
+	LongviewGrant     types.Set    `tfsdk:"longview_grant"`
+	NodebalancerGrant types.Set    `tfsdk:"nodebalancer_grant"`
+	StackscriptGrant  types.Set    `tfsdk:"stackscript_grant"`
+	VolumeGrant       types.Set    `tfsdk:"volume_grant"`
+	DatabaseGrant     types.Set    `tfsdk:"database_grant"`
+	ID                types.String `tfsdk:"id"`
+}
+
+func (data *DataSourceModel) parseUser(
+	ctx context.Context, user *linodego.User,
+) diag.Diagnostics {
+	data.Username = types.StringValue(user.Username)
+	data.Email = types.StringValue(user.Email)
+	data.Restricted = types.BoolValue(user.Restricted)
+
+	sshKeys, diags := types.ListValueFrom(ctx, types.StringType, user.SSHKeys)
+	if diags.HasError() {
+		return diags
+	}
+	data.SSHKeys = sshKeys
+
+	id, err := json.Marshal(user)
+	if err != nil {
+		diags.AddError("Error marshalling json: %s", err.Error())
+		return diags
+	}
+
+	data.ID = types.StringValue(string(id))
```

Comment:
Can we use `username` for the `id` instead? Maybe simpler.

Reply:
Initially I considered doing that but then I decided to use the JSON-Marshalled id object so that there aren't two fields with the same data. What do you think?